modules/contact/test/tests/bouncing-block-contact_pdass/try_gitterle.i

```diff
@@ -0,0 +1,274 @@
+starting_point = 2e-1
+offset = 1e-2
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+  # scaling = 1.0
+[]
+
+[Mesh]
+  file = long-bottom-block-1elem-blocks.e
+  uniform_refine = 2
+  patch_update_strategy = always
+[]
+
+[Variables]
+  [disp_x]
+    block = '1 2'
+  []
+  [disp_y]
+    block = '1 2'
+  []
+  [normal_lm]
+    block = 3
+    use_dual = true
+  []
+  [gap_variable]
+    block = 3
+    use_dual = false
+  []
+  [vel_variable]
+    block = 3
+    use_dual = false
+  []
+  [tangential_lm]
+    block = 3
+    use_dual = true
+  []
+[]
+
+[ICs]
+  [disp_y]
+    block = 2
+    variable = disp_y
+    value = '${fparse starting_point + offset}'
+    type = ConstantIC
+  []
+[]
+
+[Modules/TensorMechanics/Master]
+  [all]
+    strain = FINITE
+    generate_output = 'stress_xx stress_yy'
+    block = '1 2'
+  []
+[]
+
+[Materials]
+  [elasticity_2]
+    type = ComputeIsotropicElasticityTensor
+    block = '2'
+    youngs_modulus = 1e3
+    poissons_ratio = 0.3
+  []
+  [elasticity_1]
+    type = ComputeIsotropicElasticityTensor
+    block = '1'
+    youngs_modulus = 1e6
+    poissons_ratio = 0.3
+  []
+  [stress]
+    type = ComputeFiniteStrainElasticStress
+    block = '1 2'
+  []
+[]
+
+[Constraints]
+  [weighted_gap_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+  []
+
+ # Create gap_variable
+  [weighted_gap_indicator]
+    type = ComputeWeightedGapLMMechanicalContactIndicator
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = gap_variable
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+  []
+
+ # Create vel_variable
+  [tangential_vel]
+    type = ComputeWeightedTangentialVelocityLMMechanicalContactIndicator
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = vel_variable
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+    contact_pressure = normal_lm
+  []
```

Comment:
@lindsayad  Here, I am using these "redundant" computations of weighted gap and tangential velocities. Just to be used to define the actual frictional constraints based on PDASS.
I guess one drawback of doing this is losing derivative information coming from the the mortar constraint. You mentioned you had an alternative way for doing this based on an object that loops over all the mortar segments contributing to a secondary node.